torch/_refs/__init__.py

```diff
@@ -860,54 +855,59 @@ def trunc(a):
 
 
 def _make_elementwise_binary_reference(
-    prim: Callable,
-    *,
     type_promotion_kind,
     aten_op=infer_aten_op,
+    name=None,
     has_out=True,
     supports_lhs_python_scalar=True,
     supports_rhs_python_scalar=True,
     supports_two_python_scalars=False,
 ) -> Callable:
-    @elementwise_type_promotion_wrapper(
-        type_promoting_args=("a", "b"),
-        type_promotion_kind=type_promotion_kind,
-    )
-    def _ref(
-        a: Union[Tensor, NumberType],
-        b: Union[Tensor, NumberType],
-    ) -> Tensor:
-        if not supports_lhs_python_scalar and isinstance(a, Number):
-            raise ValueError(
-                "Received a lhs Python scalar to an elementwise binary operation that does not accept lhs scalars!"
-            )
+    def inner(prim: Callable):
```

Comment:
What's going on with this change?

Reply:
This just aligns `_make_elementwise_binary_reference` with `_make_elementwise_unary_reference` in that, given some parameters, it returns a decorator.

Reply:
I also moved from if nots to checks, improve the error messages (priting the correct name) and I set the name of the `ref`